lib/hook.js

```diff
@@ -174,6 +175,39 @@ function unhookRunInThisContext() {
     vm.runInThisContext = originalRunInThisContext;
 }
 
+/**
+ * hooks `vm.runInNewContext` to return transformed code.
+ * @method hookRunInNewContext
+ * @static
+ * @param matcher {Function(filePath)} a function that is called with the filename passed to `vm.runInNewContext`
+ *  Should return a truthy value when transformations need to be applied to the code, a falsy value otherwise
+ * @param transformer {Function(code, filePath)} a function called with the original code and the filename passed to
+ *  `vm.createScript`. Should return the transformed code.
+ * @param options {Object} options Optional.
+ * @param {Boolean} [options.verbose] write a line to standard error every time the transformer is called
+ */
+function hookRunInNewContext(matcher, transformer, opts) {
+    opts = opts || {};
+    var fn = transformFn(matcher, transformer, opts.verbose);
+    vm.runInNewContext = function (code, context, file) {
+        var ret = fn(code, file);
+        Object.keys(global).forEach(function(key) {
+          if (key.indexOf('$$cov_') === 0) {
```

Comment:
and I needed populate `opt.injectScope` before I call the `hook`?
```javascript
...
opts.injectScope = {};
Object.keys(global).forEach(function(key) {
   if (key.indexOf('$$cov_') === 0) {
      opts.injectScope[key] = global[key];
   }
});
istanbul.hook.hookRunInNewContext(matcher, transformer, opts);
```
how I will know if the keys have not changed?
It has some function that gives me the keys?